### ISSUE TITLE
Teensy 4.1 crashing when receiving 0-Length packets 

### ISSUE THREAD

**Issue by stavgafny** [vjmuzik/NativeEthernet#38]
### Description:
On the Teensy 4.1, using the following minimal setup code, the application crashes when it receives a UDP packet with a length of 0 bytes:

```cpp
void setup()
{
    Ethernet.begin(macAddress, ipAddress);
    Udp.begin(address.port);
}

void loop() {}
```

### Steps to Reproduce:

1. Upload the minimal setup code to a Teensy 4.1 board.
2. Send a UDP packet with 0 bytes of data to the board.

The expected behavior is for it to receive the packet with 0 length in size of its data.
The actual behavior is that the application crashes. 

The crash might be due to not validating packets with a data length of 0 bytes or incorrect handling of such packets.

**Comment by ssilverman** [comment]
Note that the _[QNEthernet](https://github.com/ssilverman/QNEthernet)_ library supports zero-length UDP packets.

**Comment by stavgafny** [comment]
Yeah turns out QNEthernet is the recommended library for working with teensy4.1
